custom_model_runner/datarobot_drum/drum/language_predictors/r_predictor/r_predictor.py

```diff
@@ -104,3 +104,42 @@ def predict(self, input_filename):
                 logger.error(error_message)
                 raise DrumCommonException(error_message)
         return py_data_object
+
+    def predict_unstructured(self, **kwargs):
+        def _r_is_character(r_val):
+            _is_character = ro.r("is.character")
+            return bool(_is_character(r_val))
+
+        def _rlist_to_dict(rlist):
+
+            dd = {}
+            for k, v in rlist.items():
+                k = str(k)
+                if k == UnstructuredDtoKeys.DATA:
+                    if v == ro.rinterface.NULL:
+                        dd[k] = None
+                    else:
+                        dd[k] = str(v[0]) if _r_is_character(v) else bytes(v)
+                else:
+                    # Any scalar value is returned from R as one element vector,
+                    # so get this value.
+                    dd[k] = str(v[0]) if v != ro.rinterface.NULL else None
+            return dd
+
+        data_binary_or_text = kwargs.get(UnstructuredDtoKeys.DATA, None)
+
+        if UnstructuredDtoKeys.DATA in kwargs:
+            kwargs.pop(UnstructuredDtoKeys.DATA)
+
+        # if data_binary_or_text is str it will be auto converted into R character type;
+        # otherwise if it is bytes, manually convert it into byte vector (raw)
+        r_data_binary_or_text = data_binary_or_text
+        if isinstance(data_binary_or_text, bytes):
+            r_data_binary_or_text = ro.rinterface.ByteSexpVector(data_binary_or_text)
+
+        kwargs_filtered = {k: v for k, v in kwargs.items() if v is not None}
+        kwargs_filtered[UnstructuredDtoKeys.DATA] = r_data_binary_or_text
+        ret_rlist = r_handler.predict_unstructured(model=self._model, **kwargs_filtered)
```

Comment:
To be honest since data is the only key now, i would probably move it out of kwargs since most users will want to use it right away. What do you think?

I mean I would have then the main interface be score_unstructured(model, body, **kwargs) in both Python and R

Reply:
***DONE***